### PR TITLE
Bump doctrine/doctrine-bundle and sonata-project/datagrid-bundle

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because it should be done in master.

## Changelog

```markdown
### Changed
- Bump "doctrine/doctrine-bundle" to ^2.0
- Bump "sonata-project/datagrid-bundle" to ^3.0
### Removed
- Removed support for "swiftmailer/swiftmailer-bundle"
```

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -3,3 +3,7 @@ APP_SECRET=13345f5bb5c82aa1cc0033b64cde8989
 
 MAILER_URL=smtp://mailhog:1025
 DATABASE_URL=mysql://sonata:sonata@mysql:3306/sonata?serverVersion=5.7
+
+###> symfony/mailer ###
+# MAILER_DSN=smtp://localhost
```
Comment (VincentLanglet) [inline .env:8]: You have both a MAILER_URL and a MAILER_DSN

File: config/bundles.php [hunk 2]
```diff
@@ -1,24 +1,12 @@
...
-
-/*
- * This file is part of the Sonata Project package.
- *
- * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
```
Comment (OskarStark) [inline config/bundles.php:13]: ?

**Comment by wbloszyk** [comment]
After do it I got some error like:
```
 Argument 1 passed to Sonata\Component\Product\SeoProductIterator::__construct() must be an instance of Symfony\Bridge\Doctrine\RegistryInterface, instance of Doctrine\Bundle\DoctrineBundle\Registry given, called in /home/zbiru/Phps  
  tormProjects/sandbox_3/var/cache/dev/ContainerN2fiw4U/srcApp_KernelDevDebugContainer.php on line 17529                          
```
and
```
Argument 3 passed to Sonata\ProductBundle\Block\RecentProductsBlockService::__construct() must be an instance of Symfony\Bridge\Doctrine\RegistryInterface, instance of Doctrine\Bundle\DoctrineBundle\Registry given, called in /home/zbiru/PhpstormProjects/sandbox_3/var/cache/dev/ContainerFehTqnf/srcApp_KernelDevDebugContainer.php on line 9953
```

Both are in ecommerce and are made by inject `<argument type="service" id="doctrine"/>`. WDYT We should drop support for `doctrine/doctrine-bundle v1` and present it as potential BC-break or we should change `RegistyInterface` to `ManagerRegistry`? @greg0ire 

Also thanks for @OskarStark and @VincentLanglet for very quick review. :)

File: config/bundles.php (hunk 2 above)
Comment (wbloszyk) [reply config/bundles.php]: Done

File: .php_cs.cache [hunk 3]
```diff
@@ -1 +1 @@
...
...onStartCommandTest.php":2218374630,"tests\/Sonata\/Bundle\/QABundle\/SonataNotificationCleanUpCommandTest.php":375822255,"tests\/Sonata\/Bundle\/QABundle\/SonataCacheFlushAllCommandTest.php":1609900566,"tests\/Sonata\/Bundle\/QABundle\/SonataPageUpdateCoreRouteCommandTest.php":4096342370,"tests\/Sonata\/Bundle\/QABundle\/SonataNewsSyncCommentTest.php":1235694350,"tests\/Sonata\/Bundle\/QABundle\/ListCommandTest.php":1617049927,"tests\/Sonata\/Bundle\/QABundle\/SonataNotificationRestartCommandTest.php":4170258971,"tests\/Sonata\/Bundle\/QABundle\/SonataNotificationListQueuesCommandTest.php":1360509770,"tests\/Sonata\/Bundle\/QABundle\/CommandTestCase.php":482749113,"tests\/Sonata\/Bundle\/QABundle\/SonataSeoSitemapCommandTest.php":952352379,"tests\/Sonata\/Bundle\/QABundle\/SonataCacheFlushCommandTest.php":2266785432,"tests\/Sonata\/Bundle\/QABundle\/SonataMediaSyncCommandTest.php":2689791940,"tests\/Sonata\/Bundle\/QABundle\/SonataEasyExtendsDumpMappingCommandTest.php":395709145,"tests\/Sonata\/Bundle\/QABundle\/SonataNotificationHandlerListCommandTest.php":2706085264,"tests\/Sonata\/Bundle\/QABundle\/SonataPageRenderBlockCommandTest.php":3402949528,"public\/index.php":3066197350}}
```
Comment (VincentLanglet) [inline .php_cs.cache:1]: Shouldn't we add this to .gitignore ?

**Comment by greg0ire** [comment]
Dropping support for doctrine bundle v1 is not a BC break if that is the question.

File: .env (hunk 1 above)
Comment (wbloszyk) [reply .env:8]: It was added by recipies. `MAILER_URL` is in swiftmailer config. We do not need both?

**Comment by wbloszyk** [comment]
> Dropping support for doctrine bundle v1 is not a BC break if that is the question.

I do not ask about it. Anyway all of sonata bundles suport `doctrine/doctrine-bundle v2` so It is even nice idea to drop support for 1.x. Are you agree? @greg0ire

File: config/packages/mailer.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+framework:
+    mailer:
+        dsn: '%env(MAILER_DSN)%'
```
Comment (VincentLanglet) [inline config/packages/mailer.yaml:3]: You're using MAILER_DSN, but you removed it from the .env in favor of MAILER_URL.

There is something not clear for me.

File: .env (hunk 1 above)
Comment (OskarStark) [reply .env:8]: Then you can remove the section, can't you?

**Comment by OskarStark** [comment]
Can you finish this one @wbloszyk ?

And please rebase, thanks

File: .env (hunk 1 above)
Comment (wbloszyk) [reply .env:8]: I will keep this section becouse we are changing swiftmailer to mailer:
```
wbloszyk@wbloszyk-Z170-D3H:~/PhpstormProjects/sandbox_3$ composer why symfony/mailer
sonata-project/user-bundle  4.10.2  requires  symfony/mailer (^4.4 || ^5.1)  
```
```
wbloszyk@wbloszyk-Z170-D3H:~/PhpstormProjects/sandbox_3$ composer why symfony/swiftmailer-bundle
sonata-project/sandbox      dev-master  requires  symfony/swiftmailer-bundle (^3.4)  
sonata-project/news-bundle  3.16.0      requires  symfony/swiftmailer-bundle (^3.4)  
```

Proof: https://github.com/sonata-project/SonataUserBundle/commit/ecc961b15218627ee1106f2f53679fcc9ef27efd

@phansys Can you do the same for SonataNewsBundle?

Comment (wbloszyk) [reply .env:8]: BTW we should also change `MAILER_URL` to the `MAILER_DNS` like it is using in mailer.

Comment (phansys) [reply .env:8]: >@phansys Can you do the same for SonataNewsBundle?

I'll try to find time this week.

Comment (wbloszyk) [reply .env:8]: @phansys https://github.com/sonata-project/SonataNewsBundle/pull/711

**Comment by wbloszyk** [comment]
RTM

**Comment by VincentLanglet** [comment]
All the test are failing @wbloszyk

**Comment by wbloszyk** [comment]
@VincentLanglet The test will be taking becouse we do not rewrite it to new system.  They are from symfony 2.

**Comment by VincentLanglet** [comment]
I don't understand. First things to do would be to fix the tests.
Linter is failing
Cache:clear is returning an error:
```
You have requested a non-existent parameter "sonata.comment.class.comment.e  
!!    ntity". Did you mean this: "sonata.news.admin.comment.entity"?      
```

I cant make a proper review with a failing ci which throw a lot of errors.
Maybe someone else thinks differently.

**Comment by wbloszyk** [comment]
@VincentLanglet The tests are always failing from time when I update symfony from 2 to 4. Look at tests in other PRs, like:
https://github.com/sonata-project/sandbox/pull/1004

> ```
> You have requested a non-existent parameter "sonata.comment.class.comment.e  
> !!    ntity". Did you mean this: "sonata.news.admin.comment.entity"?      
> ```

This error was introduce in SonataCommentBundle 3.3. I check it with 3.2 and it was disapere.

BTW
Can you also look at: https://github.com/sonata-project/sandbox/issues/634

**Comment by VincentLanglet** [comment]
> @VincentLanglet The tests are always failing from time when I update symfony from 2 to 4. Look at tests in other PRs, like:
> #1004

Yes, so fixing the tests should be the priority. How can we bump dependencies without tests ?
If it break something we cant know it.
 
> > ```
> > You have requested a non-existent parameter "sonata.comment.class.comment.e  
> > !!    ntity". Did you mean this: "sonata.news.admin.comment.entity"?      
> > ```
> 
> This error was introduce in SonataCommentBundle 3.3. I check it with 3.2 and it was disapere.

This should come from https://github.com/sonata-project/SonataCommentBundle/blob/55d0784bc7325d1a644fa5f2a1cfe39c0fdb3d5f/src/DependencyInjection/SonataCommentExtension.php#L315

It was introduced in https://github.com/sonata-project/SonataCommentBundle/pull/190, maybe @jordisala1991 could help

**Comment by wbloszyk** [comment]
> I cant make a proper review with a failing ci which throw a lot of errors.
> Maybe someone else thinks differently.

We agree to accept errors in test to allow upgrade to symfony 4 directly from:
https://github.com/sonata-project/sandbox/blob/2.3/composer.json

We are make it sandbox:master working but from this time I do not have time to finish it.
 
> This should come from https://github.com/sonata-project/SonataCommentBundle/blob/55d0784bc7325d1a644fa5f2a1cfe39c0fdb3d5f/src/DependencyInjection/SonataCommentExtension.php#L315
> 
> It was introduced in [sonata-project/SonataCommentBundle#190](https://github.com/sonata-project/SonataCommentBundle/pull/190), maybe @jordisala1991 could help

It is getter. Anyway I working on it.

**Comment by jordisala1991** [comment]
https://github.com/sonata-project/SonataCommentBundle/blob/55d0784bc7325d1a644fa5f2a1cfe39c0fdb3d5f/src/DependencyInjection/SonataCommentExtension.php#L131

The parameter should be defined here. Even thought it could be workinh with 3.2 , the logic there didnt change a lot.

**Comment by jordisala1991** [comment]
Oh, it is used previous its definition. Probably the bug is there.

**Comment by wbloszyk** [comment]
> Oh, it is used previous its definition. Probably the bug is there. Can you check @wbloszyk ?

Correct, now it is working. Do you have time for make PR or should I do it?

**Comment by jordisala1991** [comment]
> > Oh, it is used previous its definition. Probably the bug is there. Can you check @wbloszyk ?
> 
> 
> 
> Correct, now it is working. Do you have time for make PR or should I do it?

Better if you do it, I can review it.

**Comment by wbloszyk** [comment]
I merge this PR becouse tests are working on my local machine afer some changes in vendor. I describe require changes in #1008

**Pull request merged**